spec/requests/api/v1/partners_requests_spec.rb

```diff
@@ -48,6 +48,85 @@
     end
   end
 
+  describe "PUT /api/v1/partners" do
+    let(:headers) { {'X-Api-Key': ENV["DIAPER_KEY"]} }
+
+    context "when we set the partner to pending" do
+      let(:partner) { create(:partner) }
+      let(:params) { { partner: { diaper_partner_id: partner.diaper_bank_id, status: "pending" }} }
+
+      it 'returns OK' do
+        valid_partner_update_request(params, headers)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'sets the status to pending' do
+        valid_partner_update_request(params, headers)
+        expect(partner.reload.partner_status).to eq("pending")
+      end
+    end
+
+    context "when we set the partner to recertification_required" do
+      let(:partner) { create(:partner) }
+      let(:params) { { partner: { diaper_partner_id: partner.diaper_bank_id, status: "recertification_required" }} }
+
+      it 'returns OK' do
+        valid_partner_update_request(params, headers)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'sets the status to Recertification Required' do
+        valid_partner_update_request(params, headers)
+        expect(partner.reload.partner_status).to eq("Recertification Required")
+      end
+
+      it 'shows recertification required status in the response body' do
+        valid_partner_update_request(params, headers)
+        expect(JSON.parse(response.body)['message']).to eq('Partner status: Recertification Required.')
+      end
+    end
+
+    context "when we set the partner to approved" do
+      let(:partner) { create(:partner) }
+      let(:params) { { partner: { diaper_partner_id: partner.diaper_bank_id, status: "approved" }} }
+
+      it 'returns OK' do
+        valid_partner_update_request(params, headers)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'sets the status to Verified' do
+        valid_partner_update_request(params, headers)
+        expect(partner.reload.partner_status).to eq("Verified")
```

Comment:
I think @kdreyeroren Is doing some work to ensure that we are using consistent `"snake_case"` strings for partner_status. We can address as a merge conflict later.

Reply:
ok